### DISCUSSION TITLE
Legal: Permissions to use the name/brand 'typst'

### DISCUSSION THREAD

**Discussion by victor-sm** [typst/typst#375] [General]
There are already [70 repositories](https://github.com/search?q=typst&type=repositories) containing typst in their name. I think it would be important for the team/company to clarify in which contexts the usage is permitted. As far as I understood, typst will have some form of paid product in the future and the company also bears the same name, which means (at least in Europe) that using it could result in legal consequences.

Some form of codifications would also be important to prevent any disruption should the company be sold or the team change.

Here are some questions off the top of my head:

- Can I name my library typst-xyz? Or should it be called xyz-for-typst for example?
- If in the future alternative editors get developed, are they allowed to contain typst in their name?
- What about other products, libraries, services, or templates? What if they are commercial in nature?
- The same questions could by applied to the logo I think.

**Comment by rpitasky** [comment]
This is definitely an important question for the founders @laurmaedje and @reknih, pinging them so they see this.

**Comment by Kreijstal** [comment]
isn't it open source?

**Reply by victor-sm** [reply]
A software license generally doesn't affect the trademark/branding/naming rights. The Apache 2.0 license used by this project even explicitly states this under point 6:

> 6. Trademarks. This License does not grant permission to use the trade
   names, trademarks, service marks, or product names of the Licensor,
   except as required for reasonable and customary use in describing the
   origin of the Work and reproducing the content of the NOTICE file.

**Comment by reknih** [comment]
Hey, this is an important question, so I'll try to answer it here:

- You are free to use the Typst name and logo to refer to [our web app](https://typst.app) and the Typst compiler.
- Commercial uses of name and logo always require our authorization.
- You may use the Typst name in the name of your package or project that in some way incorporates, depends on, extends, or references Typst. The name of the project must not imply that it is the Typst web app / IDE or compiler. The name must allow any reader to immediately grasp that the project is neither the Typst web app nor compiler.
    - **OKAY:** `typst-din-letters`, `typst-vim-plugin`
    - **NOT OKAY:** `typst-ide`, `typst-app`, `typstc`, `typst`
 - You may create and use modified versions of the Typst logo or its elements (a) for a project that in some way incorporates, depends on, extends, or references Typst and (b) on blogs / social media / ... as long as they are substantial enough to not be confused for the original logo. Given your modified logo, a consumer shall not assume that it designates the Typst compiler or web app, nor shall they assume that it is endorsed by us.
   - **OKAY:** Mashing up the logo on social media or on your blog, Incorporating the Typst "t" in your larger app logo, referring to the web app or the compiler with the logo as-is
   - **NOT OKAY:** Using the unmodified logo or the Typst "t" as the logo for your project.
- If you are unsure about whether your use of the Typst logo or name is permissible or if you want to obtain explicit permission or exemptions from these rules, contact us at hello@typst.app

This is in line with how other projects like [Python](https://www.python.org/psf/trademarks/) handle their name and logo. Note that this policy is an initial version that may evolve in the future.

**Reply by rpitasky** [reply]
Can this be posted somewhere more official, i.e. a branding guidelines page or in the TOS or something?

**Reply by laurmaedje** [reply]
https://typst.app/legal/brand/

**Comment by ghost** [comment]
I think no one has registered the name, at least in Germany where Typst seems to be.
Some software logos or names are officially protected, like [the Python logo](https://register.dpma.de/DPMAregister/marke/registerHABM?AKZ=012595575&CURSOR=30) or the [Mozilla](https://register.dpma.de/DPMAregister/marke/registerHABM?AKZ=000645820&CURSOR=1) name.

I'm not an expert here, but if I remember correctly, a name/brand can be used freely in countries where there is no protection registered for this country.

**Reply by victor-sm** [reply]
In Germany, since typst is the name of the company _Typst GmbH_ they already have pretty good protections.

**Reply by ghost** [reply]
Yes, the company name is protected when the company is registered.
But the product name is a different field, I think.

**Reply by victor-sm** [reply]
Yes, but in this case it doesn't matter since according to §15 Abs. 2 MarkenG, it would have the same protections.

**Reply by ghost** [reply]
I think this is still different from a product name, see [here](https://www.mittelstand-südwestfalen.de/recht-und-steuern/gewerblicher-rechtsschutz/detail/artikel/11777-geschaeftliche-bezeichnung-vs-marke-gleiche-rechte). But I'm not an expert.

**Reply by victor-sm** [reply]
I don't think this edge case would apply, since typsts would not be local, but quite the opposite. Anyhow, I hope the team make sure the name get's some international protection if they want to be safe long term, since there might be another edge case just around the corner.
